contracts/utils/BatchTransferNFT.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+/**
+ * @title BatchTransferNFT
+ * @notice Enables to batch transfer multiple NFTs in a single call to this contract
+ */
+contract BatchTransferNFT {
+    struct Transfer {
+        address nft;
+        address recipient;
+        uint256 tokenId;
+        uint256 amount;
+    }
+
+    /**
+     * @notice Batch transfer different NFT in a single call
+     * @param _transfers The list of transfer.
+     * The amount defines the type of NFT:
+     *  - amount = 0: ERC721
+     *  - amount > 0: ERC1155
+     */
+    function batchTransfer(Transfer[] calldata _transfers) external {
+        uint256 _length = _transfers.length;
+        unchecked {
+            for (uint256 i; i < _length; ++i) {
+                Transfer memory _transfer = _transfers[i];
+                if (_transfer.amount == 0) {
+                    IERC721(_transfer.nft).safeTransferFrom(
+                        msg.sender,
+                        _transfer.recipient,
+                        _transfer.tokenId,
+                        ""
```

Comment:
nit: I think you can just do `.safeTransferFrom` w/ 3 args and it will be the same (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol#L164-L170) but up to you

Reply:
Very interesting, but the safeTransferFrom with 3 args costs less than the one with 4, I would have expected the opposite as the 3 args calls the 4 args and adding the string